packages/did-resolver-interface/src/models/resolver.ts

```diff
@@ -42,30 +43,12 @@ export interface IServiceEndpoint {
   type: string;
   serviceEndpoint: string;
   description?: string;
-  validity: BigNumber;
-  block: number;
+  validity?: BigNumber;
+  block?: number;
```

Comment:
Could you please elaborate why these properties has become optional?

Reply:
The reason these are made optional was because, there are `deleted` in https://github.com/energywebfoundation/ew-did-registry/blob/56744f7ec6376f43868774d28796c210b4bbaf62/packages/did-ethr-resolver/src/functions/functions.ts#L379

With the `typescript` version update, it has become more strict and the `build` throws error 
"src/functions/functions.ts(385,14): error TS2790: The operand of a 'delete' operator must be optional."

Reply:
I think that the actual error is how public key is constructed in `wrapDidDocument`. `validity` and `block` are in fact required fields

Reply:
Could you please elaborate more on the implementation side, why exactly we are deleting the values before pushing to the `didDocument`?

Reply:
Removed the delete property call from `wrapDpcument` method.

Reply:
validity and block are removed because they is no point to keep them in the document. This confusion appeared because `IPublicKey` is used for implementation (blockchain) details and for representation, which was not the best decision